includes/admin/discounts/discount-actions.php

```diff
@@ -204,35 +204,35 @@ function edd_admin_edit_discount( $data = array() ) {
 	// Start date.
 	if ( ! empty( $data['start_date'] ) ) {
 		$start_date        = sanitize_text_field( $data['start_date'] );
-		$start_date_hour   = (int) $data['start_date_hour'] >= 0 && (int) $data['start_date_hour'] <= 23
-			? sanitize_text_field( $data['start_date_hour'] )
+		$start_date_hour   = isset( $data['start_date_hour'] ) && (int) $data['start_date_hour'] >= 0 && (int) $data['start_date_hour'] <= 23
+			? intval( $data['start_date_hour'] )
 			: '00';
-		$start_date_minute = (int) $data['start_date_minute'] >= 0 && (int) $data['start_date_minute'] <= 59
-			? sanitize_text_field( $data['start_date_minute'] )
+		$start_date_minute = isset( $data['start_date_minute'] ) && (int) $data['start_date_minute'] >= 0 && (int) $data['start_date_minute'] <= 59
+			? intval( $data['start_date_minute'] )
 			: '00';
 
 		// The start date is entered in the user's WP timezone. We need to convert it to UTC prior to saving now.
```

Comment:
@ashleyfae testing this with a new discount and the start/end dates for which I've left the times empty, have been converted to:
- `2020-12-15 05:00:00`
- `2020-12-19 04:59:00`
Which I believe is intentional as I read the code, but I'm also seeing that existing discounts are set to `00:00:00` and `23:59:59` respectively. The existing discounts were migrated from 2.9.x, so if this conversion is correct, maybe it needs to be applied on migrated discounts as well?

Reply:
Sorry my steps are actually kind of deceptive.

In the database times are always stored in UTC. My steps were deceptive because my time zone is currently the same as UTC, which means what I enter is exactly how it gets saved. **This is not the case for other time zones.**

In your case, here's the expected result:

1. It defaults to start at midnight _your time zone_ and end at 23:59 _your time zone_. 
2. It then gets converted to UTC when it gets saved in the database. So that means it will add 5 hours.

Therefore what you're seeing is correct behaviour; my testing notes were just kind of crappy.

---

Regarding the imported ones -- that depends on how you entered them to begin with. We've always stored discounts in UTC, but until now, we never did the time zone conversion. That means technically in 2.9 you should be entering start and end times _in UTC_ -- not in your local time. So if you entered them as 00:00 in 2.9, then seeing them saved now as 00:00 is correct.

Reply:
Thanks, @ashleyfae, that makes sense. This is technically outside the scope of this PR, but I'm wondering if it might make more sense to _save_ the user entered time and convert to the site time on use/retrieval display. I can think of some instances where the current scenario could result in inaccurate times:
- When I set up a discount with specific times before a time change happens, but the discount dates are after the time change (so what I enter will be an hour off, as I understand it).
- If I change my site time zone (unusual/unlikely, but if I discover that my site has incorrectly been running in the default time zone and I change it, then the discount will have been saved in a different time than what's being used).

Reply:
Saving in UTC is pretty standard for all time-related things and I don't think we're going to deviate from that.